source/write/bulk-write.txt

```diff
@@ -0,0 +1,287 @@
+   :language: python
+   :copyable:
+
+If any of the write operations fail, {+driver-short+} raises a
+``BulkWriteError`` and does not perform any further operations.
+``BulkWriteError`` provides a ``details`` attribute that includes the operation
+that failed, and details about the exception.
+
+.. note::
+
+   When PyMongo runs a bulk operation, it uses the ``write_concern`` of the
+   collection in which the operation is running. The driver reports all write
+   concern errors after attempting all operations, regardless of execution order. 
+
+Customize Bulk Write Operations
+-------------------------------
+
+The ``bulk_write()`` method optionally accepts additional
+parameters, which represent options you can use to configure the bulk write
+operation. If you don't specify any additional options, the driver does not customize
+the bulk write operation.
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``ordered``
+     - | If ``True``, the driver performs the write operations in the order
+         provided. If an error occurs, the remaining operations are not
+         attempted. If ``False``, the driver performs the operations in an
+         arbitrary order and attempts to perform each operation.
+       | Defaults to ``True``.
+
+   * - ``bypass_document_validation``
+     - | Specifies whether the operation bypasses document-level validation. For more
+         information, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the MongoDB
+         Server manual.
+       | Defaults to ``False``.
+
+   * - ``session``
+     - | An instance of ``ClientSession``. For more information, see the `API
+         documentation <{+api-root+}pymongo/client_session.html#pymongo.client_session.ClientSession>`__.
+
+   * - ``comment``
+     - | A comment to attach to the operation. For more information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
+         MongoDB Server manual for more information.
+
+   * - ``let``
+     - | A map of parameter names and values. Values must be constant or closed
+         expressions that don't reference document fields. For more information,
+         see the :manual:`let statement
+         </reference/command/delete/#std-label-delete-let-syntax>` in the
+         MongoDB Server manual.
+
+The following example uses the ``bulk_write()`` with the ``ordered`` options set
+to ``False``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-write-unordered
+   :end-before: end-bulk-write-unordered
+   :language: python
+   :copyable:
+
+If any of the write operations in the preceding example fail, {+driver-short+}
+reports the errors only after attempting all operations.
+
+.. note::
+
+   Unordered bulk operations do not guarantee order of execution. The order can
+   differ from the way you list them to optimize the runtime.
```

Comment:
Earlier it says "arbitrary" order; does it actually do runtime optimization here?